tests/show_techsupport/test_techsupport.py

```diff
@@ -0,0 +1,253 @@
+        with loganalyzer:
+            teardown_acl(duthost, acl_setup)
+
+
+# MIRRORING PART #
+
+@pytest.fixture(scope='function')
+def neighbor_ip(duthost, testbed):
+    # ptf-32 topo is not supported in mirroring
+    if testbed['topo']['name'] == 'ptf32':
+        pytest.skip('Unsupported Topology')
+
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    dst_ip = mg_facts["minigraph_portchannel_interfaces"][0]['peer_addr']
+    yield str(dst_ip)
+
+
+@pytest.fixture(scope='function')
+def mirror_setup(duthost):
+    """
+    setup fixture
+    """
+    logger.debug("creating running directory ...")
+    duthost.command('mkdir -p {}'.format(MIRROR_RUN_DIR))
+    tmp_path = duthost.tempfile(path=MIRROR_RUN_DIR, state='directory', prefix='mirror', suffix="")['path']
+
+    setup_info = {
+        'dut_tmp_dir': tmp_path,
+    }
+    logger.info('setup variables {}'.format(pprint.pformat(setup_info)))
+    yield setup_info
+
+
+@pytest.fixture(scope='function')
+def gre_version(duthost):
+    asic_type = duthost.facts['asic_type']
+    if asic_type in ["mellanox"]:
+        SESSION_INFO['gre'] = 0x8949 # Mellanox specific
+    else:
+        SESSION_INFO['gre'] = 0x6558
+
+
+@pytest.fixture(scope='function')
+def mirroring(duthost, neighbor_ip, mirror_setup, gre_version):
+    """
+    fixture gathers all configuration fixtures
+    :param duthost: dut host
+    :param mirror_setup: mirror_setup fixture
+    :param mirror_config: mirror_config fixture
+    """
+
+    logger.info("Adding mirror_session to dut")
+    acl_rule_file = os.path.join(mirror_setup['dut_tmp_dir'], ACL_RULE_PERSISTENT_FILE)
+    extra_vars = {
+        'acl_table_name':  EVERFLOW_TABLE_NAME,
+    }
+    logger.info('extra variables for MIRROR table:\n{}'.format(pprint.pformat(extra_vars)))
+    duthost.host.options['variable_manager'].extra_vars.update(extra_vars)
+
+    duthost.template(src=os.path.join(TEMPLATE_DIR, ACL_RULE_PERSISTENT_J2), dest=acl_rule_file)
+    duthost.command('config mirror_session add {} {} {} {} {} {} {}'
+    .format(SESSION_INFO['name'], SESSION_INFO['src_ip'], neighbor_ip,
+     SESSION_INFO['dscp'], SESSION_INFO['ttl'], SESSION_INFO['gre'], SESSION_INFO['queue']))
+
+    logger.info('Loading acl mirror rules ...')
+    load_rule_cmd = "acl-loader update full {} --session_name={}".format(acl_rule_file, SESSION_INFO['name'])
+    duthost.command('{}'.format(load_rule_cmd))
+
+    try:
+        yield
+    finally:
+        loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix='acl')
+        loganalyzer.load_common_config()
+
+        try:
+            loganalyzer.expect_regex = [LOG_EXCEPT_MIRROR_SESSION_REMOVE]
+            with loganalyzer:
+                teardown_mirroring(duthost, mirror_setup['dut_tmp_dir'])
+        except LogAnalyzerError as err:
+            raise err
+
+
+def teardown_mirroring(dut, tmp_path):
+    """
+    teardown EVERFLOW rules after test by applying empty configuration
```

Comment:
Same here. Is it correct to assume no rules were applied on any topology before the test? Shouldn't we just remove what was added during the test?